tests/components/togrill/test_config_flow.py

```diff
@@ -0,0 +1,142 @@
+"""Test the ToGrill config flow."""
+
+from unittest.mock import Mock
+
+from bleak.exc import BleakError
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant import config_entries
+from homeassistant.components.togrill.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from . import TOGRILL_SERVICE_INFO, TOGRILL_SERVICE_INFO_NO_NAME
+
+from tests.components.bluetooth import inject_bluetooth_service_info
+
+pytestmark = pytest.mark.usefixtures("mock_setup_entry")
+
+
+async def test_user_selection(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test we can select a device."""
+
+    inject_bluetooth_service_info(hass, TOGRILL_SERVICE_INFO)
+    inject_bluetooth_service_info(hass, TOGRILL_SERVICE_INFO_NO_NAME)
+    await hass.async_block_till_done(wait_background_tasks=True)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={"address": "00000000-0000-0000-0000-000000000001"},
+    )
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["data"] == {
+        "address": "00000000-0000-0000-0000-000000000001",
+        "model": "Pro-05",
+        "probe_count": 0,
+    }
+    assert result["title"] == "Pro-05"
+    assert result["result"].unique_id == "00000000-0000-0000-0000-000000000001"
+
+
+async def test_failed_connect(
+    hass: HomeAssistant,
+    mock_client: Mock,
+    mock_client_class: Mock,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test we can select a device."""
+
+    mock_client_class.connect.side_effect = BleakError("Failed to connect")
+
+    inject_bluetooth_service_info(hass, TOGRILL_SERVICE_INFO)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={"address": "00000000-0000-0000-0000-000000000001"},
```

Comment:
would be nice if we can use the constant for `address`

Reply:
Done